### DISCUSSION TITLE
Does 'pyright' support the `$/progress` LSP protocol?

### DISCUSSION THREAD

**Discussion by linrongbin16** [microsoft/pyright#6818] [Q&A]
I'm a Neovim plugin developer, my Neovim plugin doesn't receive progress notifications from pyright via the `$/progress` method (see issue: https://github.com/linrongbin16/lsp-progress.nvim/issues/113).

Does pyright support `$/progress` protocol method? or there's some other relate protocols?

**Comment by erictraut** [comment]
Progress for which operation? Pyright will send progress status to the client if it is type checking all of the files in a project (i.e. if `diagnosticMode` is set to "workspace" rather than the default of "openFilesOnly"). It doesn't send progress for analysis of a single file because type analysis of a single file typically tens of milliseconds, maybe hundreds of milliseconds for really large files.

In any case, we have no plans to add additional progress updates in pyright.
